File: asm.25.lab4/clients/akm2501/st14/console_io.py
FIELD_LABELS = {
    "name": "Имя",
    "age": "Возраст",
    "department": "Отдел",
    "team_size": "Размер команды",
    "has_company_car": "Служебный автомобиль",
}

TYPE_HINTS = {
    str: "строка",
    int: "целое число",
    float: "число",
    bool: "да/нет",
}


class ConsoleIO:
    def _label(self, field_name: str) -> str:
        return FIELD_LABELS.get(field_name, field_name)

    def _convert(self, text, to_type):
        if to_type is None:
            return text

        if to_type is bool:
            t = text.strip().lower()
            return t in ("1", "true", "t", "y", "yes", "да", "д", "истина")

        try:
            return to_type(text)
        except Exception:
            return text

    def _type_hint(self, field_type) -> str:
        if field_type is None:
            return ""
        return TYPE_HINTS.get(field_type, field_type.__name__)

    def _format_value(self, value):
        if isinstance(value, bool):
            return "да" if value else "нет"
        if value is None or value == "":
            return "—"
        return str(value)

    def read_field(self, obj, field_name):
        expected_type = None
        if hasattr(obj, "FIELDS"):
            expected_type = obj.FIELDS.get(field_name)

        label = self._label(field_name)
        hint = self._type_hint(expected_type)
        prompt = f"Введите {label}"
        if hint:
            prompt += f" ({hint})"
        prompt += ": "

        raw = input(prompt)

        try:
            value = self._convert(raw, expected_type)
            setattr(obj, field_name, value)
        except Exception:
            print("Ошибка: неверный формат. Поле не изменено.")

    def write_field(self, obj, field_name):
        value = getattr(obj, field_name, None)
        label = self._label(field_name)
        print(f"{label}: {self._format_value(value)}")

File: asm.25.lab5/Lab5/akm2501/st14/console_io.go
package akm250114

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type ConsoleIO struct {
	reader *bufio.Reader
}

func NewConsoleIO() *ConsoleIO {
	return &ConsoleIO{reader: bufio.NewReader(os.Stdin)}
}

var fieldLabels = map[string]string{
	"name":            "Имя",
	"age":             "Возраст",
	"department":      "Отдел",
	"team_size":       "Размер команды",
	"has_company_car": "Служебный автомобиль",
}

func (io *ConsoleIO) label(fieldName string) string {
	if label, ok := fieldLabels[fieldName]; ok {
		return label
	}
	return fieldName
}

func (io *ConsoleIO) convert(text string, fieldType string) interface{} {
	text = strings.TrimSpace(text)

	switch fieldType {
	case "bool":
		t := strings.ToLower(text)
		return t == "да" || t == "д" || t == "yes" || t == "y" || t == "true" || t == "1"
	case "int":
		if val, err := strconv.Atoi(text); err == nil {
			return val
		}
		return 0
	default:
		return text
	}
}

func (io *ConsoleIO) formatValue(value interface{}) string {
	if value == nil {
		return "—"
	}
	switch v := value.(type) {
	case bool:
		if v {
			return "да"
		}
		return "нет"
	case string:
		if v == "" {
			return "—"
		}
	}
	return fmt.Sprintf("%v", value)
}

func (io *ConsoleIO) inputString(prompt string) string {
	fmt.Print(prompt)
	text, _ := io.reader.ReadString('\n')
	return strings.TrimSpace(text)
}

func (io *ConsoleIO) ReadField(emp map[string]interface{}, fieldName string) {
	label := io.label(fieldName)

	var fieldType string
	switch fieldName {
	case "name", "department":
		fieldType = "string"
	case "age", "team_size":
		fieldType = "int"
	case "has_company_car":
		fieldType = "bool"
	}

	var prompt string
	if fieldType == "bool" {
		prompt = fmt.Sprintf("%s (да/нет): ", label)
	} else {
		prompt = fmt.Sprintf("Введите %s: ", label)
	}

	text := io.inputString(prompt)
	value := io.convert(text, fieldType)
	emp[fieldName] = value
}

func (io *ConsoleIO) WriteField(emp map[string]interface{}, fieldName string) {
	label := io.label(fieldName)
	value := emp[fieldName]
	fmt.Printf("%s: %s\n", label, io.formatValue(value))
}
